pkg/filters/auth.go

```diff
@@ -53,6 +54,20 @@ func WithAuthentication(
 			return
 		}
 
+		// http2 is an expensive protocol that is prone to abuse,
+		// see CVE-2023-44487 and CVE-2023-39325 for an example.
+		// Do not allow unauthenticated clients to keep these
+		// connections open (i.e. basically degrade them to the
+		// performance of http1 with keep-alive disabled).
+		//
+		// https://github.com/kubernetes/apiserver/blob/master/pkg/endpoints/filters/authentication.go#L107C1-L117C4
+		if req.ProtoMajor == 2 && isAnonymousUser(res.User) {
+			// limit this connection to just this request,
+			// and then send a GOAWAY and tear down the TCP connection
+			// https://github.com/golang/net/commit/97aa3a539ec716117a9d15a4659a911f50d13c3c
+			w.Header().Set("Connection", "close")
```

Comment:
aren't we supposed to return here? you are continuing the handler chain.

Reply:
This closes the TCP connection. The inflight-request will be handled and then the client needs to re-do the TCP handshake, as far as I understood it.

It actually decreases the http2 speed for unauthenticated users to worse then http/1.1 as they need to go through the TCP handshake process.